lbrynet/core/StreamDescriptor.py

```diff
@@ -445,7 +445,10 @@ def download_sd_blob(session, blob_hash, payment_rate_manager, timeout=None):
                                           payment_rate_manager,
                                           session.wallet,
                                           timeout)
+    mirror = HTTPBlobDownloader(session.blob_manager, [blob_hash], session.download_mirrors)
+    mirror.start()
     sd_blob = yield downloader.download()
+    mirror.stop()
```

Comment:
What if the mirror finishes first? I think this should cancel whichever didn't finish. You could have the `start` and `download` calls in a deferred list set to fire on one callback

Reply:
It is done implicitly over blob manager. When one finishes a blob, the other will see that through the shared blob manager and go to the next (or halt, if nothing else to do). I can add that as a class documentation on the `HTTPBlobDownloader` class.